python/ray/autoscaler/cli_logger.py

```diff
@@ -0,0 +1,426 @@
+import sys
+
+import click
+
+import colorama
+from colorful.core import ColorfulString
+import colorful as cf
+colorama.init()
+
+def _tostr(x):
+    if not isinstance(x, str):
+        return str(x)
+    return x
+
+def _strip_codes(msg):
+    return msg # todo
+
+# we could bold "{}" strings automatically but do we want that?
+# todo:
+def _format_msg(msg, *args, **kwargs):
+    if isinstance(msg, str) or isinstance(msg, ColorfulString):
+        tags_str = ""
+        if "_tags" in kwargs:
+            tags_list = []
+            for k, v in kwargs["_tags"].items():
+                if v is True:
+                    tags_list += [k]
+                    continue
+                if v is False:
+                    continue
+
+                tags_list += [k+"="+v]
+            if tags_list:
+                tags_str = cf.reset(
+                    cf.gray(" [{}]".format(", ".join(tags_list))))
+
+            del kwargs["_tags"]
+
+        numbering_str = ""
+        if "_numbered" in kwargs:
+            chars, i, n = kwargs["_numbered"]
+
+            i = _tostr(i)
+            n = _tostr(n)
+
+            numbering_str = cf.gray(
+                chars[0] + i + "/" + n + chars[1]) + " "
+
+            del kwargs["_numbered"]
+
+        no_format = False
+        if "_no_format" in kwargs:
+            no_format = kwargs["_no_format"]
+            del kwargs["_no_format"]
+
+        if no_format:
+            # todo: throw if given args/kwargs?
+            return numbering_str + msg + tags_str
+        return numbering_str + msg.format(*args, **kwargs) + tags_str
+
+    if kwargs:
+        raise ValueError("We do not support printing kwargs yet.")
+
+    l = [msg, *args]
+    l = [_tostr(x) for x in l]
+    return ', '.join(l)
+
+class _CliLogger():
+    def __init__(self):
+        self.strip = False
+        self.old_style = True
+        self.color_mode = "auto"
+        self.indent_level = 0
+        self.verbosity = 0
+
+        self.dump_command_output = False
```

Comment:
doesn't seem like this is implemented anywhere?

Reply:
it's set in each command (`create_or_update_cluster` and `teardown_cluster`)